op-conductor/rpc/ws/flashblocks_handler.go

```diff
@@ -0,0 +1,407 @@
+		h.rollupBoostConn.Close()
+		h.rollupBoostConn = nil
+	}
+
+	// Force close the HTTP server if it's running
+	if h.server != nil {
+		h.log.Info("closing WebSocket server")
+		err := h.server.Close()
+		if err != nil {
+			h.log.Error("Error closing WebSocket server", "err", err)
+		}
+		h.log.Info("WebSocket server closed")
+	}
+}
+
+// BroadcastMessage sends a message to all connected WebSocket clients
+func (h *Handler) BroadcastMessage(message []byte) {
+	h.log.Trace("Broadcasting message to WebSocket clients")
+	h.hub.broadcast <- message
+}
+
+// listenToRollupBoost listens for messages from the rollup boost WebSocket
+func (h *Handler) listenToRollupBoost(ctx context.Context) {
+	defer func() {
+		if h.rollupBoostConn != nil {
+			h.log.Info("closing rollup boost WebSocket connection")
+			h.rollupBoostConn.Close()
+			h.rollupBoostConn = nil
+		}
+	}()
+
+	retryCount := 0
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-h.hub.done:
+			return
+		default:
+			// Check if connection is nil and try to reconnect
+			if h.rollupBoostConn == nil {
+				if retryCount >= MaxReconnectAttempts {
+					h.log.Error("exceeded maximum reconnection attempts to rollup boost WebSocket", "maxRetries", MaxReconnectAttempts)
+					return
+				}
+
+				h.log.Info("attempting to connect to rollup boost WebSocket", "url", h.cfg.RollupBoostWsURL, "attempt", retryCount+1)
+				conn, _, err := websocket.DefaultDialer.Dial(h.cfg.RollupBoostWsURL, nil)
+				if err != nil {
+					retryCount++
+					h.log.Warn("failed to connect to rollup boost WebSocket, will retry", "err", err, "retryIn", ReconnectDelay)
+					time.Sleep(ReconnectDelay)
+					continue
+				}
+
+				h.rollupBoostConn = conn
+				h.log.Info("successfully reconnected to rollup boost WebSocket")
+				retryCount = 0 // Reset retry counter on successful connection
+			}
+
+			// Read messages from the connection
+			_, message, err := h.rollupBoostConn.ReadMessage()
+			if err != nil {
+				h.log.Warn("error reading from rollup boost WebSocket", "err", err)
+				// Close the connection and try to reconnect
+				h.rollupBoostConn.Close()
+				h.rollupBoostConn = nil
+				continue
+			}
+
+			h.handleRollupBoostMessage(ctx, message)
+		}
+	}
+}
+
+// handleRollupBoostMessage processes a message received from rollup boost
+func (h *Handler) handleRollupBoostMessage(ctx context.Context, message []byte) {
+	// Only forward messages if we're the leader - check dynamically each time
+	if !h.isLeaderFn(ctx) {
+		h.log.Debug("not forwarding rollup boost message, not the leader")
+		return
+	}
+
+	// Forward the message to connected clients
+	h.BroadcastMessage(message)
+}
+
+// startWebSocketServer initializes and starts a WebSocket server
+func (h *Handler) startWebSocketServer(_ context.Context) error {
+	if h.cfg.WebsocketServerPort <= 0 {
+		h.log.Info("WebSocket server disabled, no port configured")
+		return nil
+	}
+
+	// Create the hub for this WebSocket server
+	h.hub = newHub()
```

Comment:
Let's create a new abstraction called `WebSocketServer`? imo `hub` should be part of it, and we can handle the start / init / stop functions together.